core/src/main/java/org/mskcc/cbio/oncokb/cache/CacheFetcher.java

```diff
@@ -244,6 +249,42 @@ public Alteration getAlterationFromGenomeNexus(GNVariantAnnotationType gnVariant
         return AlterationUtils.getAlterationFromGenomeNexus(gnVariantAnnotationType, genomicLocation, referenceGenome);
     }
 
+    public void cacheAlterationFromGenomeNexus(GenomeNexusAnnotatedVariantInfo gnAnnotatedVariantInfo) {
+        Alteration alteration = new Alteration();
+
+        // Build the Alteration object from the pre-annotated GN variant object.
+        if (StringUtils.isNotEmpty(gnAnnotatedVariantInfo.getHugoSymbol())) {
+            Gene gene = GeneUtils.getGene(gnAnnotatedVariantInfo.getHugoSymbol());
+            if (gene == null) {
+                gene = new Gene();
+                gene.setHugoSymbol(gnAnnotatedVariantInfo.getHugoSymbol());
+                gene.setEntrezGeneId(gnAnnotatedVariantInfo.getEntrezGeneId());
+            }
+            alteration.setGene(gene);
+        }
+        alteration.setAlteration(gnAnnotatedVariantInfo.getHgvspShort());
+        alteration.setProteinStart(gnAnnotatedVariantInfo.getProteinStart());
+        alteration.setProteinEnd(gnAnnotatedVariantInfo.getProteinEnd());
+        alteration.setConsequence(VariantConsequenceUtils.findVariantConsequenceByTerm(gnAnnotatedVariantInfo.getConsequenceTerms()));
+
+        String hgvsg = gnAnnotatedVariantInfo.getHgvsg();
+        String genomicLocation = gnAnnotatedVariantInfo.getGenomicLocation();
+        ReferenceGenome referenceGenome = gnAnnotatedVariantInfo.getReferenceGenome();
+
+        // Store pre-annotated alteration into Redis cache
+        Cache cache = cacheManager.getCache(CacheCategory.GENERAL.getKey() + REDIS_KEY_SEPARATOR + "getAlterationFromGenomeNexus");
```

Comment:
Can we use ConcatGenerator?

Reply:
I don't think we can because ConcatGenerator only generates the part of the key with the method params. 
```
public class ConcatGenerator implements KeyGenerator {

    public Object generate(Object target, Method method, Object... params) {
        return StringUtils.arrayToDelimitedString(params, REDIS_KEY_SEPARATOR);
    }
}
```
I am fetching the cache from the cache manager with the hardcoded name `CacheCategory.GENERAL.getKey() + REDIS_KEY_SEPARATOR + "getAlterationFromGenomeNexus"`. Not sure if there's a better way to figure out which cache we should put the values into.